src/lib/y2network/dialogs/wireless_wep_keys.rb

```diff
@@ -0,0 +1,253 @@
+                  VSpacing(1)
+                ),
+                HSpacing(3)
+              )
+            ),
+            VSpacing(1)
+          ),
+          HSpacing(5)
+        )
+      end
+
+      # Always open new dialog to work properly in sequence
+      def should_open_dialog?
+        true
+      end
+
+      class WEPKeyLength < CWM::ComboBox
+        def initialize(builder)
+          textdomain "network"
+
+          @builder = builder
+        end
+
+        def items
+          [
+            ["64", "64"],
+            ["128", "128"]
+          ]
+        end
+
+        def init
+          length_s = @builder.key_length.to_s
+          self.value = length_s.empty? ? "128" : length_s
+        end
+
+        def store
+          @builder.key_length = value.to_i
+        end
+
+        def label
+          _("&Key Length")
+        end
+
+        def help
+          _(
+            "<p><b>Key Length</b> defines the bit length of your WEP keys.\n" \
+              "Possible are 64 and 128 bit, sometimes also referred to as 40 and 104 bit.\n" \
+              "Some older hardware might not be able to handle 128 bit keys, so if your\n" \
+              "wireless LAN connection does not establish, you may need to set this\n" \
+              "value to 64.</p>"
+          )
+        end
+      end
+
+      class WEPKeys < CWM::CustomWidget
+        def initialize(settings)
+          textdomain "network"
+          @settings = settings
+        end
+
+        def contents
+          VBox(
+            Table(
+              Id(:wep_keys_table),
+              Header(
+                # Table header label
+                # Abbreviation of Number
+                _("No."),
+                # Table header label
+                _("Key"),
+                # Table header label
+                Center(_("Default"))
+              )
+            ),
+            HBox(
+              # PushButton label
+              PushButton(Id(:wep_keys_add), Yast::Label.AddButton),
+              # PushButton label
+              PushButton(Id(:wep_keys_edit), Yast::Label.EditButton),
+              # PushButton label
+              PushButton(Id(:wep_keys_delete), Yast::Label.DeleteButton),
+              # PushButton label
+              PushButton(Id(:wep_keys_default), _("&Set as Default"))
+            )
+          )
+        end
+
+        # TODO: help text which explain format of WEP keys
+
+        def init
+          refresh_table
+        end
+
+        def refresh_table
+          table_items = @settings.keys.each_with_index.map do |key, i|
+            next unless key
+            Item(Id(i), i.to_s, key, i == @settings.default_key ? "X" : "")
+          end
+
+          Yast::UI.ChangeWidget(Id(:wep_keys_table), :Items, table_items.compact)
+          [:wep_keys_delete, :wep_keys_edit, :wep_keys_default].each do |k|
+            Yast::UI.ChangeWidget(
+              Id(k),
+              :Enabled,
+              !@settings.keys.compact.empty?
```

Comment:
why not put `@setting.keys.compact` into its own variable? Just to avoid calling to `#compact` twice.

Reply:
actually it is called four times due to this each. I will do it.